app/controlplane/internal/service/group.go

```diff
@@ -479,6 +479,58 @@ func (g *GroupService) UpdateMemberMaintainerStatus(ctx context.Context, req *pb
 	return &pb.GroupServiceUpdateMemberMaintainerStatusResponse{}, nil
 }
 
+// ListProjects retrieves a paginated list of projects that a group is a member of.
+func (g *GroupService) ListProjects(ctx context.Context, req *pb.GroupServiceListProjectsRequest) (*pb.GroupServiceListProjectsResponse, error) {
+	currentOrg, err := requireCurrentOrg(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse orgID
+	orgUUID, err := uuid.Parse(currentOrg.ID)
+	if err != nil {
+		return nil, errors.BadRequest("invalid", "invalid organization ID")
+	}
+
+	// Parse groupID and groupName from the request
+	id, name, err := req.GetGroupReference().Parse()
+	if err != nil {
+		return nil, errors.BadRequest("invalid", fmt.Sprintf("invalid group reference: %s", err.Error()))
+	}
+
+	// Initialize the options for getting projects
+	groupOpts := &biz.ListProjectsByGroupOpts{
+		IdentityReference: &biz.IdentityReference{
+			ID:   id,
+			Name: name,
+		},
+		VisibleProjectsIDs: g.visibleProjects(ctx),
```

Comment:
I'd rather call this FilterByProject since it can be empty and that means that the filter is empty, not that there are not projects

```suggestion
		VisibleProjectsIDs: g.visibleProjects(ctx),
```